### PR TITLE
refactor: Simplify Deposit test + helpers

### PR DESCRIPTION
**Pull request by pxrl**
Simplify deposit generation by converting getDepositParams() to accept an object, rather than a fixed list of parameters. This makes it cleaner for the caller to opt-out of supplying various inputs.

This was prompted by a need to update the deposit() and getDepositParams() utilities such that the caller can supply a custom message.

### CODE REVIEW

File: test/fixtures/SpokePool.Fixture.ts [hunk 1]
```diff
@@ -74,22 +82,23 @@ export async function deposit(
...
-        recipient.address,
-        token.address,
-        amountToDeposit,
-        destinationChainId,
-        depositRelayerFeePct,
-        quoteTimestamp ?? (await spokePool.getCurrentTime())
-      )
-    );
+  quoteTimestamp ??= await spokePool.getCurrentTime();
+  await spokePool.connect(depositor).deposit(
```
Comment (pxrl) [inline test/fixtures/SpokePool.Fixture.ts:91]: This looks incorrect - it should probably use `BigNumber.toNumber()` instead.

File: test/fixtures/SpokePool.Fixture.ts [hunk 2]
```diff
@@ -116,12 +125,12 @@ export async function fillRelay(
...
-  depositAmount: BigNumber = consts.amountToDeposit,
-  amountToRelay: BigNumber = consts.amountToRelay,
-  realizedLpFeePct: BigNumber = consts.realizedLpFeePct,
-  relayerFeePct: BigNumber = consts.depositRelayerFeePct
+  depositId = consts.firstDepositId,
+  originChainId = consts.originChainId,
+  depositAmount = consts.amountToDeposit,
+  amountToRelay = consts.amountToRelay,
+  realizedLpFeePct = consts.realizedLpFeePct,
+  relayerFeePct = consts.depositRelayerFeePct
```
Comment (pxrl) [inline test/fixtures/SpokePool.Fixture.ts:133]: Opportunistic cleanup here.

File: test/fixtures/SpokePool.Fixture.ts [hunk 3]
```diff
@@ -222,24 +231,25 @@ export function getRelayHash(
...
-    _maxCount ? _maxCount.toString() : consts.maxUint256.toString(),
+    args.recipient ?? randomAddress(),
+    args.originToken,
+    args.amount.toString(),
+    args.destinationChainId.toString(),
+    args.relayerFeePct.toString(),
+    args.quoteTimestamp.toString(),
+    args.message ?? "0x",
+    args?.maxCount?.toString() ?? consts.maxUint256.toString(),
   ];
```
Comment (pxrl) [inline test/fixtures/SpokePool.Fixture.ts:253]: This was the entry point for this change - we need to be able to generate deposits with a custom message.

File: test/fixtures/SpokePool.Fixture.ts (hunk 1 above)
Comment (pxrl) [reply test/fixtures/SpokePool.Fixture.ts:90]: Fixed: https://github.com/across-protocol/contracts-v2/pull/363/commits/217dc9ea8508e0e90f440c5ddfd97a4d9c8b875f

**Review by nicholaspai — APPROVED** [review]
LGTM thanks for doing this. We had a TODO in the recent clean up to copy these utilities to the relayer repo. Should we do it now?

**Comment by pxrl** [comment]
> LGTM thanks for doing this. We had a TODO in the recent clean up to copy these utilities to the relayer repo. Should we do it now?

It's not clear to me whether it's better to copy, or just invest effort in fixing the utils here. If we copied then it'd probably have to go to sdk-v2, since we also depend on the functionality there. I'm just not a big fan of having duplicates of these utilities.

**Pull request merged**